Adding Dockerfile for providing REST API via Docker container

IntegralProgrammer opened the pull request:
NeuralCR is now capable of running in a Docker container and providing a REST API.

IntegralProgrammer committed:
Adding Dockerfile for providing REST API via Docker container

sdumitriu commented on README.md:
````diff
@@ -150,6 +150,31 @@ A web app is available for NCR trained on HPO:
+Build the Docker container:
+
+```bash
+docker build -t docker-neuralcr .
````
```suggestion
docker build -t ccmsk/neuralcr .
```

IntegralProgrammer replied:
Fixed :heavy_check_mark:

sdumitriu commented on ncrmodel.py:
```diff
@@ -3,7 +3,8 @@
 import json
 import pickle 
-import fastText
+#import fastText
```
```suggestion
```

IntegralProgrammer replied:
Fixed :heavy_check_mark:

sdumitriu commented on train.py:
```diff
@@ -4,7 +4,8 @@
 from onto import Ontology
 import json
-import fastText
+#import fastText
```
```suggestion
```

IntegralProgrammer replied:
Fixed :heavy_check_mark:

IntegralProgrammer committed:
Revise according to PR feedback

IntegralProgrammer committed:
Add ability to select different trained models through REST API

Add ability to list/delete trained models through REST API

Add ability to generate HPF QSUB jobs for model training

Bugfixes in model training code

IntegralProgrammer committed:
Fix incorrect syntax for print() statement

Redefine print() statement so that log entries are written to a file (not STDOUT) - needed as a workaround for a bug on HPF

IntegralProgrammer committed:
Switch OBO parser to Orange

Now works with MONDO ontology

Lock pip package versions

IntegralProgrammer committed:
First commit of CI test cases for NCR REST API

IntegralProgrammer committed:
Switched from using Pickle to serialize the parsed Ontology to using JSON - this is much safer as Pickle allows for arbitrary code execution

IntegralProgrammer committed:
Data-driven test template prints errors instead of exiting with a non-zero exit code

IntegralProgrammer committed:
Updating codebase to be compatible with the publicly available trained Tensorflow models (HP:0000118 and MONDO:0045024)

Fixed bug when checking for ~/ncr_obo_uploads and ~/ncr_qsub - exits as soon as any one of these directories do not exist.

IntegralProgrammer committed:
Update README.md with content on running CI tests

IntegralProgrammer committed:
Added support for auto test execution through the AUTOTEST env var

Automatically downloads trained models from GitHub if the directory /root/opt/ncr/model_params does not exist in the container
